### PR TITLE
Adding workflow name to workflow crd labels

### PR DESCRIPTION
**Pull request by surindersinghp**
We need this information so we can get pod level usage metrics for workflows. I will be adding a similar change later to plugins repo for array job.

### CODE REVIEW

File: pkg/compiler/transformers/k8s/workflow.go [hunk 1]
```diff
@@ -171,6 +171,7 @@ func BuildFlyteWorkflow(wfClosure *core.CompiledWorkflowClosure, inputs *core.Li
 	if err != nil {
 		errs.Collect(errors.NewWorkflowBuildError(err))
 	}
+	obj.ObjectMeta.Labels[WorkflowIDLabel] = primarySpec.ID
```
Comment (EngHabu) [inline pkg/compiler/transformers/k8s/workflow.go]: There is a limit on the length of label fields, beyond which the creation of the CRD will fail... we should probably add code to guard against that.

Comment (EngHabu) [inline pkg/compiler/transformers/k8s/workflow.go]: Is it really the full `ID`? or just the `Name` part of the ID?

Comment (surindersinghp) [reply pkg/compiler/transformers/k8s/workflow.go]: fixed, it was more to check if this is the right place to inject this.

Comment (surindersinghp) [reply pkg/compiler/transformers/k8s/workflow.go]: thanks for asking. It was project:domain:name, changed it to name. Added a simple test.

**Comment by surindersinghp** [comment]
> Can you summarize the motivation behind the change?
We need this information so we can pod level usage metrics for workflows. I will be adding a similar change later to plugins repo for array job.
@EngHabu

File: pkg/compiler/transformers/k8s/workflow.go [hunk 2]
```diff
@@ -171,6 +181,7 @@ func BuildFlyteWorkflow(wfClosure *core.CompiledWorkflowClosure, inputs *core.Li
 	if err != nil {
 		errs.Collect(errors.NewWorkflowBuildError(err))
 	}
+	obj.ObjectMeta.Labels[WorkflowIDLabel] = utils.SanitizeLabelValue(WorkflowNameFromID(primarySpec.ID))
```
Comment (EngHabu) [inline pkg/compiler/transformers/k8s/workflow.go]:
```suggestion
	obj.ObjectMeta.Labels[WorkflowNameLabel] = utils.SanitizeLabelValue(WorkflowNameFromID(primarySpec.ID))
```

File: pkg/compiler/transformers/k8s/workflow.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/lyft/flytepropeller/pkg/compiler/common"
```
Comment (EngHabu) [inline pkg/compiler/transformers/k8s/workflow.go]: Also note, you will need to `dep ensure -update` in flyte admin to pull in this change

File: pkg/controller/nodes/node_exec_context.go [hunk 4]
```diff
@@ -96,9 +100,19 @@ func (e execContext) MaxDatasetSizeBytes() int64 {
...
+	return e.nodeLabels
+}
+
 func newNodeExecContext(_ context.Context, store *storage.DataStore, w v1alpha1.ExecutableWorkflow, node v1alpha1.ExecutableNode, nodeStatus v1alpha1.ExecutableNodeStatus, inputs io.InputReader, maxDatasetSize int64, er events.TaskEventRecorder, tr handler.TaskReader, nsm *nodeStateManager, enqueueOwner func() error) *execContext {
+	md := execMetadata{WorkflowMeta: w}
+	nodeLabels := md.GetLabels()
+	if nodeLabels == nil {
+		nodeLabels = make(map[string]string)
+	}
+	nodeLabels[NodeIDLabel] = utils.SanitizeLabelValue(node.GetID())
```
Comment (EngHabu) [inline pkg/controller/nodes/node_exec_context.go]: Should we also only put the node name here? or do you need the full id? the id will include the execution id as well?

File: pkg/compiler/transformers/k8s/workflow.go (hunk 2 above)
Comment (surindersinghp) [reply pkg/compiler/transformers/k8s/workflow.go]: fixed

File: pkg/compiler/transformers/k8s/workflow.go (hunk 3 above)
Comment (surindersinghp) [reply pkg/compiler/transformers/k8s/workflow.go]: yep!

File: pkg/controller/nodes/node_exec_context.go (hunk 4 above)
Comment (surindersinghp) [reply pkg/controller/nodes/node_exec_context.go]: it is just the node name.

**Comment by surindersinghp** [comment]
@EngHabu regarding adding taskname to labels in taskHandler instead of plugin executor (in plugins repo).

Labels are encoded in NodeExecutionMetadata which TaskHandler just passes it down to plugins (via tastcontext https://github.com/lyft/flytepropeller/blob/master/pkg/controller/nodes/task/handler.go#L347). NodeExecContext doesnt have set/update label apis. I think its best we add task label in plugins executor and keep task handler simpler.
